### ISSUE TITLE
PathItems not stable in rendered out document

### ISSUE THREAD

**Issue by TristanSpeakEasy** [pb33f/libopenapi#109]
When adding new path items to a document, the order of these will change from run to run (ie not deterministically ordered).

Would be good if these are ordered.

Longer term I wonder what you think about moving to something like https://pkg.go.dev/github.com/wk8/go-ordered-map (which I believe kin-openapi has switched to) to maintain the order of maps items in the openapi document,

but in the shorter term maybe sorting the keys of new items added to maps to try and make output deterministic?

**Comment by TristanSpeakEasy** [comment]
So this is specifically for any new items added to a document that already has items

**Comment by pquerna** [comment]
I'm also constructing new Documents from scratch in memory (no previous documents to load) in https://github.com/ductone/protoc-gen-apigw (new project, much WIP).

For both PathItems, and Component.Schemas (and others), what if as a final Sort() of the items -- if Line number is Unset or Equal, sort by the Path?

Currently in the Paths/MarshalYAML there is a block:
```
func (p *Paths) MarshalYAML() 
....
	for k, pi := range p.PathItems {
		ln := 9999 // default to a high value to weight new content to the bottom.
		if p.low != nil {
			lpi := p.low.FindPath(k)
			if lpi != nil {
				ln = lpi.ValueNode.Line
			}
		}
		mapped = append(mapped, &pathItem{pi, k, ln, nil})
	}

...

	sort.Slice(mapped, func(i, j int) bool {
		return mapped[i].line < mapped[j].line
	})
```

If the sort slice was changed to something like:
```
	sort.Slice(mapped, func(i, j int) bool {
                if mapped[i].line == mapped[j].line {
                       return mapped[i].path < mapped[j].path
                }
		return mapped[i].line < mapped[j].line
	})
```

I think the output would be stable-ish?  much more so than today.  The ideas in #119 is fine too, but maybe is a much heavier lift than some strategic sorting?

**Comment by daveshanley** [comment]
This has been resolved in `v0.14`
